builtin/logical/pki/path_acme_order.go

```diff
@@ -797,28 +807,59 @@ func parseOrderIdentifiers(data map[string]interface{}) ([]*ACMEIdentifier, erro
 			return nil, fmt.Errorf("value argument for value in 'identifiers' can not be blank: %w", ErrMalformed)
 		}
 
+		identifier := &ACMEIdentifier{
+			Type:          ACMEDNSIdentifier,
+			Value:         valueStr,
+			OriginalValue: valueStr,
+		}
+
 		switch typeStr {
 		case string(ACMEIPIdentifier):
 			ip := net.ParseIP(valueStr)
 			if ip == nil {
 				return nil, fmt.Errorf("value argument (%s) failed validation: failed parsing as IP: %w", valueStr, ErrMalformed)
 			}
-			identifiers = append(identifiers, &ACMEIdentifier{
-				Type:  ACMEIPIdentifier,
-				Value: valueStr,
-			})
+			identifiers = append(identifiers, identifier)
```

Comment:
Is this not wrong, why are we switching the identifier to an DNS identifier here?

Reply:
Ah, yes.